### PR TITLE
Add preliminary CLI

### PR DESCRIPTION
**Pull request by danlamanna**
This is definitely not in the state that I want it, but I've been holding off pushing it for so long that I wanted to get something out. The name `rvt` is arbitrary.

At the moment there are some dummy credentials hardcoded since I wasn't sure what the approach to take with authentication was. 

`rvt sync dkc://<folder-id> ./some/local/path` (download) should work, but upload is half implemented so far.
`rvt ls dkc://<folder-id>` should work, even with the fancy `--tree` option.

### CODE REVIEW

**Comment by zachmullen** [comment]
@danlamanna could you put the core changes into a separate PR? It seems like they're small and should be easy to approve independently.

File: rvt/rvt/__init__.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+def _maybe_upload_file(ctx, rfolder: RemoteFolder, lpath: Path):
+    rfile = rfolder.file_by_name(ctx, lpath.name)
+
+    if lpath.exists() and rfile and lpath.stat().st_mtime == rfile.modified.timestamp():
+        logger.debug(f'skipping file {lpath} (same mtime).')
+        ctx.skipped_files.append(rfile)
+        return
+
+    logger.info(f'uploading {lpath}')
+    with open(lpath, 'wb') as stream:
```
Comment (jbeezley) [inline rvt/rvt/__init__.py:340]:
```suggestion
    with open(lpath, 'rb') as stream:
```

**Comment by danlamanna** [comment]
Future work:
- [ ] figure out the best way to test this
- [ ] add a `cp` command for individual files
- [ ] add a `configure` command for persisting credentials (might make a click extension for this)

**Comment by danlamanna** [comment]
@jbeezley can you verify sync, ls, and ls-tree work the way you would expect? then we can get a basic version of this merged.

**Comment by brianhelba** [comment]
You'll need to rebase this before merging.

**Pull request closed without merging**